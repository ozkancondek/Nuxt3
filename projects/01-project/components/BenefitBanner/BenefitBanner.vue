<template>
  <div
    class="
      tw-bg-white tw-rounded tw-border tw-border-gray-300 tw-p-4 tw-pt-6
      lg:tw-px-8 lg:tw-py-6
    "
    data-track-section="benefit_banner"
  >
    <Headline
      v-if="headline.content.text"
      v-bind="headline.content"
      size="M"
      class="tw-mb-6"
    ></Headline>
    <div
      class="tw-flex tw-flex-wrap tw-gap-2.5 tw-justify-between tw-items-center"
    >
      <div
        v-for="(benefit, index) in benefits"
        :key="compKey + index"
        class="
          benefit-item
          tw-flex tw-items-center tw-w-full
          lg:tw-w-auto
          tw-mb-4
          lg:tw-mb-0
        "
      >
        {{ "d" }}
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Headline from "~/components/magazin/Headline/Headline.vue";

export default defineComponent({
  name: "BenefitBanner",
  components: {
    Headline,
  },
  props: {
    headline: {
      type: Object,
      default: () => {},
    },

    benefits: {
      type: Array,
      default: () => [],
    },
  },
  setup() {
    const compKey =
      "benefits-banner-" + (Math.random() + 1).toString(36).substring(7) + "-";
    return { compKey };
  },
});
</script>

<style scoped>
@media screen and (min-width: 960px) {
  .benefit-item {
    max-width: 30%;
  }
}
</style>
