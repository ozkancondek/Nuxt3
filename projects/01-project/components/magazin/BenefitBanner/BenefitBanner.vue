<template>
  <div>
    {{ message }}
  </div>
</template>

<script lang="ts">
export default defineComponent({
  name: "BenefitBanner",
  components: {},

  props: {
    message: {
      type: String,
      default: "World",
    },
  },
});
</script>

<style scoped>
@media screen and (min-width: 960px) {
  .benefit-item {
    max-width: 30%;
  }
}
</style>
