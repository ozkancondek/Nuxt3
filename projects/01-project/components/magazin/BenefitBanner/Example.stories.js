import Example from "./BenefitBanner.vue";

export default {
  title: "Example Folder/Example",
  component: Example,
  argTypes: {},
};

const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { Example },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return {
      args,
    };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<Example v-bind="args" />',
});

export const Primary = Template.bind({});
Primary.args = {
  text: "here is example text",
};
