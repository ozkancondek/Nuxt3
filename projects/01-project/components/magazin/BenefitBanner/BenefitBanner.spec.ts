import { mount } from "@vue/test-utils";
import BenefitBanner from "./BenefitBanner.vue";

 

test("renders BenefitBanner  correctly", () => {
  const message = "Happy People";
  const wrapper = mount(BenefitBanner, {
    props: { message },
  });
  expect(wrapper.element).toMatchSnapshot();
});
