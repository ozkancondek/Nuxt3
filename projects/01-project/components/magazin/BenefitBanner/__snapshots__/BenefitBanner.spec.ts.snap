// Vitest Snapshot v1

exports[`renders BenefitBanner  correctly 1`] = `
<div
  data-v-82b439ab=""
>
  Happy People
</div>
`;
